src/test/ui/derive-uninhabited-enum-38885.rs

```diff
@@ -5,12 +5,15 @@
 // when deriving Debug on an empty enum
 
 #[derive(Debug)]
-enum Void {} //~ WARN never used
+enum Void {}
 
 #[derive(Debug)]
-enum Foo { //~ WARN never used
+enum Foo {
     Bar(u8),
-    Void(Void),
+    Void(Void), //~ WARN never used
```

Comment:
With this change we no longer complain about unused enums, only unused variants, which might not be what we want.

Reply:
> With this change we no longer complain about unused enums, only unused variants, which might not be what we want.

Seems like a regression to me. If an enum has no variants and is never used, it should still be warned about -- similar to a struct with no fields that is never used.

Reply:
The individual variants are warned about but the enum itself isn't. I believe it should be possible to have both, but is that something we want or is it just too verbose?

Reply:
If the enum has no variants, it's certainly not too verbose to warn at all at the enum itself.

Reply:
If the enum has no variants the current code complains about the enum. Added tests showing this and other permutations.

Reply:
Thanks!